lib/backends/sdc/utils.js

```diff
@@ -163,6 +163,68 @@ function getPublishedPorts(opts, fwrules, cb) {
     });
 }
 
+
+/*
+ * Get the Docker-compatible human-readable description of the container state.
+ *
+ * This emulates:
+ * https://github.com/docker/docker/blob/483063a/container/state.go#L39-L69
+ */
+function _containerStatusFromVmobj(opts) {
+    assert.object(opts.vmobj, 'opts.vmobj');
+    assert.object(opts.log, 'opts.log');
+
+    var boot_timestamp = new Date(opts.vmobj.boot_timestamp);
+    var exittime;
+    var now = new Date();
+    var status;
+    var uptime;
+    var vmobj = opts.vmobj;
+
+    if (vmobj.state == 'running') {
+        uptime = Math.floor((now - boot_timestamp) / 1000);
+        status = 'Up ' + common.humanDuration(uptime);
+    } else if (vmobj.state == 'provisioning' && vmobj.create_timestamp) {
+        uptime = Math.floor((now - (new Date(vmobj.create_timestamp))) / 1000);
+        status = 'Provisioning ' + common.humanDuration(uptime);
+    } else if (vmobj.state == 'stopped') {
+        if (vmobj.hasOwnProperty('exit_status')
+            && vmobj.hasOwnProperty('exit_timestamp'))
+        {
+            exittime
+                = Math.floor((now - (new Date(vmobj.exit_timestamp))) / 1000);
+            status = 'Exited (' + vmobj.exit_status + ') '
+                + common.humanDuration(exittime) + ' ago';
+        } else {
+            /*
+             * Two known possibilities here:
+             * 1. The zone never started (e.g. if created, but not run; or
```

Comment:
I think this first case can be detected by the presence of docker:restartcount in the internal_metadata. If a VM has this property (even set to 0) it has been started at least once. When you do `docker create` this property is not added until the VM is started. So perhaps list as 'Created' when state==='stopped' and docker:restartcount doesn't exist?